### PR TITLE
Api text to spech

### PR DESCRIPTION
**Pull request by pakantu**
Text To Speech Component implemented in Card Page

### CODE REVIEW

File: frontend/src/components/Card.tsx [hunk 1]
```diff
@@ -1,22 +1,59 @@
...
-    return(
-        <div className="card">
-            <h3>{props.name}</h3>
-            <img className="image" src={props.image_link} alt="Ein Bild"/>
-            <ul>
-                <li>{props.animal_type}</li>
-                <li>{props.id}</li>
-            </ul>
+export default function Card(props: cardProps) {
+    const {animal_type, image_link} = props
```
Comment (Flooooooooooorian) [inline frontend/src/components/Card.tsx:16]:
```suggestion
export default function Card({animal_type, image_link}: cardProps) {
```

File: frontend/src/components/Card.tsx [hunk 2]
```diff
@@ -1,22 +1,59 @@
...
+    const [text, setText] = useState<string>('');
+    const language: string = "de-de";
+    const standardtext: string ="mit welchem Buchstaben beginnt mein Name, ich heiße";
```
Comment (Flooooooooooorian) [inline frontend/src/components/Card.tsx:19]:
```suggestion
    const standardText: string ="mit welchem Buchstaben beginnt mein Name, ich heiße";
```

File: frontend/src/components/Card.tsx [hunk 3]
```diff
@@ -1,22 +1,59 @@
...
+    //todo: set key later in environment
+    const key: string = "a7aae25de0b446c7adc2571316a7ddfc&";
```
Comment (Flooooooooooorian) [inline frontend/src/components/Card.tsx:21]: Watch out your key is public now, could be expensive if someone uses your key

File: frontend/src/components/Card.tsx [hunk 4]
```diff
@@ -1,22 +1,59 @@
...
+    /**
+     * returns a new string with the first word of the
+     * @param sentence
+     * */
+    function getFirstWord (sentence: string):string {
+        if (sentence !== undefined && sentence.includes(" ")) {
+            return sentence.split(" ")[0];
+        } else {
+            return sentence;
+        }
```
Comment (Flooooooooooorian) [inline frontend/src/components/Card.tsx:32]:
```suggestion
        return sentence.split(" ")[0];
```

File: frontend/src/components/Gallery.tsx [hunk 5]
```diff
@@ -1,38 +1,36 @@
 import Card from "./Card";
 import "./Gallery.scss"
 import {useEffect, useState} from "react";
-import {fetch10Characters, fetch1Characters} from "../services/RequestService";
+
+import {fetch1Characters} from "../services/RequestService";
 
 export default function Gallery(){
-    const [Animals10, setAnimals10] = useState<any>([]);
     const [Animals1, setAnimals1] = useState<any>([]);
```
Comment (Flooooooooooorian) [inline frontend/src/components/Gallery.tsx:8]: What does the 1 stand for?

```suggestion
    const [animals1, setAnimals1] = useState<any>([]);
```

Comment (Flooooooooooorian) [inline frontend/src/components/Gallery.tsx:8]: Define you type

File: frontend/src/context/RequireAuth.tsx [hunk 6]
```diff
@@ -8,6 +8,9 @@ export default function RequireAuth({children}: { children: ReactElement<any, an
 
     function isExpirationValid(): boolean {
         const Now = new Date();
+
```
Comment (Flooooooooooorian) [inline frontend/src/context/RequireAuth.tsx:11]: https://github.com/NiklasSchibel/AgameBC/pull/11#discussion_r797418304

File: frontend/src/pages/LoginPage.tsx [hunk 7]
```diff
@@ -62,6 +62,9 @@ export default function LoginPage() {
     };
 
     const onSubmitButton: () => void = () => {
+        console.log("This is the Username: " + usernameValue + ", for the LoginRequest")
+        console.log("Timelimit for Login: " + timevalue)
```
Comment (Flooooooooooorian) [inline frontend/src/pages/LoginPage.tsx:66]:
```suggestion

```

**Review by Flooooooooooorian — APPROVED** [review]

File: frontend/src/components/Card.tsx (hunk 1 above)
Comment (pakantu) [reply frontend/src/components/Card.tsx:16]: thanks, looks better

File: frontend/src/components/Card.tsx (hunk 3 above)
Comment (pakantu) [reply frontend/src/components/Card.tsx:21]: just a test account, but will be set in env variable next cleanup session

File: frontend/src/components/Card.tsx (hunk 4 above)
Comment (pakantu) [reply frontend/src/components/Card.tsx:32]: why delete the check?

File: frontend/src/components/Gallery.tsx (hunk 5 above)
Comment (pakantu) [reply frontend/src/components/Gallery.tsx:8]: already renamed, sorry

Comment (pakantu) [reply frontend/src/components/Gallery.tsx:8]: good point, added to issues

File: frontend/src/components/Card.tsx (hunk 4 above)
Comment (Flooooooooooorian) [reply frontend/src/components/Card.tsx:32]: sentence is of type string, so it cant be undefined.
have a look at what .split() returns when the seperator is not in the given string.

File: frontend/src/context/RequireAuth.tsx (hunk 6 above)
Comment (pakantu) [reply frontend/src/context/RequireAuth.tsx:11]: danke, deinen re-comment hatte ich nicht mehr gesehen, jetzt direkt eingebaut :)

**Pull request merged**